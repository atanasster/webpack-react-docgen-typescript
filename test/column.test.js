import compiler from './compiler.js';

test('column', async () => {
  const stats = await compiler('./examples/Column.tsx', {
    transformProps: tables => tables.map(table => ({ ...table, type: 'tsType' }))[0],
    propFilter: { 
      skipPropsWithName: ['prop4'],
    }
  });

  
  const jsonStats = stats.toJson();
  const output = jsonStats.modules[6].source;
  expect(output).toMatchSnapshot();
});
