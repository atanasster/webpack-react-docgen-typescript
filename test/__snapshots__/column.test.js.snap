// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`column 1`] = `
"import _classCallCheck from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/classCallCheck\\";
import _createClass from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/createClass\\";
import _possibleConstructorReturn from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/possibleConstructorReturn\\";
import _getPrototypeOf from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/getPrototypeOf\\";
import _inherits from \\"/Users/atanasster/webpack-react-docgen-typescript/node_modules/@babel/runtime/helpers/inherits\\";
var _jsxFileName = \\"/Users/atanasster/webpack-react-docgen-typescript/test/examples/Column.tsx\\";
import * as React from 'react';
import { Component } from 'react';
/**
 * Column properties.
 */

/**
 * Form column.
 */
export var Column =
/*#__PURE__*/
function (_Component) {
  _inherits(Column, _Component);

  function Column() {
    _classCallCheck(this, Column);

    return _possibleConstructorReturn(this, _getPrototypeOf(Column).apply(this, arguments));
  }

  _createClass(Column, [{
    key: \\"render\\",
    value: function render() {
      return React.createElement(\\"div\\", {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 25
        },
        __self: this
      }, \\"Test\\");
    }
  }]);

  return Column;
}(Component);

try {
  Column.__docgenInfo = {
    \\"description\\": \\"Form column.\\",
    \\"displayName\\": \\"Column\\",
    \\"methods\\": [],
    \\"props\\": {
      \\"prop1\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop1 description\\",
        \\"name\\": \\"prop1\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Column.tsx\\",
          \\"name\\": \\"IColumnProps\\"
        },
        \\"required\\": false,
        \\"type\\": {
          \\"name\\": \\"string\\"
        }
      },
      \\"prop2\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop2 description\\",
        \\"name\\": \\"prop2\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Column.tsx\\",
          \\"name\\": \\"IColumnProps\\"
        },
        \\"required\\": true,
        \\"type\\": {
          \\"name\\": \\"number\\"
        }
      },
      \\"prop3\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop3 description\\",
        \\"name\\": \\"prop3\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Column.tsx\\",
          \\"name\\": \\"IColumnProps\\"
        },
        \\"required\\": true,
        \\"type\\": {
          \\"name\\": \\"() => void\\"
        }
      }
    },
    \\"type\\": \\"tsType\\"
  };
} catch (e) {//eat exception
}"
`;
